src/couch_views/src/couch_views_indexer.erl

```diff
@@ -561,15 +561,19 @@ start_query_server(#mrst{qserver = nil} = Mrst) ->
         lib = Lib,
         views = Views
     } = Mrst,
-    {ok, QServer} = couch_eval:acquire_map_context(
+    case couch_eval:acquire_map_context(
             DbName,
             DDocId,
             Language,
             Sig,
             Lib,
             [View#mrview.def || View <- Views]
-        ),
-    Mrst#mrst{qserver = QServer};
+        ) of
+        {ok, QServer} ->
+            Mrst#mrst{qserver = QServer};
+        {error, Error} ->
+            throw(Error)
```

Comment:
Let's use error instead of a `throw`, usually a `throw` is used to a more expected early return here it feels more like an error.

Reply:
Done in https://github.com/cloudant/nodejs-cloudant#initialization-callback . Wdyt?